flang/lib/Lower/Allocatable.cpp

```diff
@@ -57,6 +40,33 @@ static void genAllocatableSetBounds(Fortran::lower::FirOpBuilder &builder,
   builder.create<fir::CallOp>(loc, callee, operands);
 }
 
+static void genAllocatableInitCharRtCall(Fortran::lower::FirOpBuilder &builder,
+                                         mlir::Location loc,
+                                         fir::BoxAddressValue boxAddress,
+                                         mlir::Value len) {
+  auto callee =
+      Fortran::lower::getRuntimeFunc<mkRTKey(AllocatableInitCharacter)>(
+          loc, builder);
```

Comment:
Why is this call needed? We have to allocate the space and know all the information for the descriptor. Why make an extra call to the runtime?

Reply:
It is used to set the length in the descriptor if specified in the allocate statement.
An alternative is to create a new box value could also be created with embox and be stored inside the box address before calling AllocatableAllocate.
I a may just go with this alternative. I am following the runtime API first. I will removed runtime calls that we can easily do inlined once error recovery is added and I am confident we do not need the runtime call.